edit/cms_functions.php

```diff
@@ -0,0 +1,478 @@
+    return $hidden . '<label>' . $label . ' </label><input name="value" type="text" value="' . $value . '"/>';
+}
+
+/*
+ * Called upon loading the main form section of each edit page. This retrieves the edit ID from the GET data, if one is present.
+ *
+ * @param array $getData The $_GET data we are checking.
+ *
+ * @return int Returns the ID of the entry the user wants to edit. If the user is not editing a specific entry, return minus -1.
+ */
+function getEditEntryID($getData) : int {
+    if(!key_exists('edit', $getData) ||!key_exists('id', $getData)) {
+        return -1;
+    }
+    return (int)$getData['id'];
+}
+
+
+/*
+ * Displays the form for editing projects. If there is a valid entry in the database, the values of each input field are filled out
+ * with the appropriate data.
+ *
+ * @param int $id The id of the entry to edit.
+ *
+ * @return Returns the input element for the given table, formatted as HTML. This should be placed inside the form element, before the submit
+ * element.
+ */
+function displayEditProjectInput(int $id) {
+    if($id < 1) {
+        return '';
+    }
+    $array = getProjectFromDB($id);
+    $output = '<form name="project" action="doeproject.php" method="post">';
+    $output .= '<div class="longinput"><label>Title: </label><input name="title" type="text" value="' . $array['title'] . '"></div>';
+    $output .= '<div class="longinput"><label>Type: </label><input name="type" type="text" value="' . $array['type'] . '"></div>';
+    $output .= '<div class="longinputlarge"><label>Desc: </label><textarea name="desc">' . $array['desc'] . '</textarea></div>';
+    $output .= '<div class="longinput"><label>Image: </label><input name="img" type="text" value="' . $array['image'] . '"></div>';
+    $output .= '<div class="longinput"><label>Link: </label><input name="link" type="text" value="' . $array['link'] . '"></div>';
+    $output .= '<input type="hidden" name="id" value ="' . $id . '">';
+    $output .= '<input type="submit" value="Edit"></form>';
+    return $output;
+}
+
+/*
+ * Displays the form for editing contact information. If there is a valid entry in the database, the values of each input field are filled out
+ * with the appropriate data.
+ *
+ * @param int $id The given entry to edit. This value is 0 if we are adding a new entry.
+ *
+ * @return Returns the input element for the given table, formatted as HTML. This should be placed inside the form element, before the submit
+ * element.
+ */
+function displayEditContactInfo(int $id) {
+    $array = getContactInfoFromDB($id);
+    $output = '<div class="longinput"><label>Icon ID: </label><input name="icon_id" type="text" value="' . $array['icon_id'] . '"></div>';
+    $output .= '<div class="longinput"><label>Link: </label><input name="link" type="text" value="' . $array['link'] . '"></div>';
+    $output .= '<div class="longinput"><label>Text: </label><input name="text" type="text" value="' . $array['text'] . '"></div>';
+    if($id != -1) {
```

Comment:
-1 represents a case where the user hasn't clicked on an entry in the list, and it works because -1 is an unsigned value. This is probably a habit from my self taught gamedev days where I wanted an empty comparison value for types couldn't be nullified. I think NULL will work here but I don't want to break stuff cause I'm pressed for time as it is